pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -140,12 +141,33 @@ def on_sanity_check_end(self, trainer, pl_module):
     def on_validation_end(self, trainer, pl_module):
         self._run_early_stopping_check(trainer, pl_module)
 
+    def on_train_epoch_end(self, trainer, pl_module):
+        # early stopping can also work in the train loop when there is no val loop and when using structured results
+        should_check_early_stop = False
+        train_es_key = 'early_stop_on'
+        if train_es_key in trainer.callback_metrics and trainer.callback_metrics[train_es_key] is not None:
+            self.monitor = 'early_stop_on'
+            should_check_early_stop = True
+
+        val_es_key = 'val_early_stop_on'
+        if val_es_key in trainer.callback_metrics and trainer.callback_metrics[val_es_key] is not None:
```

Comment:
shall it be rather `elif`?
```suggestion
        if trainer.callback_metrics.get(val_es_key, None) is not None:
```